R/cal-estimate-utils.R

```diff
@@ -118,6 +118,104 @@ cal_class_name.rset <- function(x) {
   "Resampled data set"
 }
 
+# ------------------------------- Data Ingestion -------------------------------
+
+
+get_tune_data <- function(x, parameters = NULL) {
+  .data <- collect_predictions(
+    x,
+    summarize = TRUE,
+    parameters = parameters
+  )
+
+  num_configs <- vctrs::vec_unique_count(.data$.config)
+  if (num_configs > 1) {
+    group <- ".config"
+  } else {
+    group <- character(0)
+  }
+
+  truth <- tune::.get_tune_outcome_names(x)
+  lvls <- levels(.data[[truth]])
+
+  if (is.null(lvls)) {
+    estimate <- ".pred"
+    lvl_map <- rlang::syms(estimate)
+    names(lvl_map) <- "predictions"
+  } else {
+    estimate <- paste0(".pred_", lvls)
+    lvl_map <- rlang::syms(estimate)
+    names(lvl_map) <- lvls
+  }
+
+  list(
+    truth = truth,
+    estimate = estimate,
+    group = group,
+    predictions = .data[, c(truth, estimate, group)],
+    levels = lvls,
+    map = lvl_map,
+    source = "tune_results"
+  )
+}
+
+get_prediction_data <- function(
+    .data,
+    truth = NULL,
+    estimate = dplyr::starts_with(".pred_"),
+    .by = NULL
+) {
+  if (!inherits(.data, "tbl_df")) {
+    data <- dplyr::as_tibble(.data)
+  }
+
+  truth <- names(tidyselect::eval_select(rlang::enquo(truth), .data))
+  estimate <- names(tidyselect::eval_select(rlang::enquo(estimate), .data))
+  by_vars <- names(tidyselect::eval_select(rlang::enquo(.by), .data))
+
+  lvls <- levels(.data[[truth]])
+
+  if (!is.null(lvls)) {
+    if (length(lvls) != length(estimate)) {
+      cli::cli_abort(
```

Comment:
This may be a regression since the previous method was permissive to have a column `.pred_class` in the selection along with the `.pred_{levels}`. 

I'm ok with this